app/ai_backend/test_genesis_connector.py

```
@@ -1523,20 +2455,22 @@ def test_unicode_boundary_conditions(self):
             '\uffff\ufffe\ufffd',  # Unicode boundaries
         ]
         
-        for test_case in unicode_test_cases:
-            with self.subTest(test_case=test_case):
-                payload = {'message': test_case}
+        for test_string in null_byte_test_cases:
```

Ruff (0.11.9)

Undefined name `null_byte_test_cases`

(F821)